rust/sdk-processor/src/utils/database.rs

```diff
@@ -0,0 +1,329 @@
+            .expect("Could not build TLS connector");
+        let connector = MakeTlsConnector::new(connector);
+
+        let (client, connection) = tokio_postgres::connect(&url, connector)
+            .await
+            .expect("Could not connect to database");
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                eprintln!("connection error: {}", e);
+            }
+        });
+        AsyncPgConnection::try_from(client).await
+    })
+    .boxed()
+}
+
+fn parse_and_clean_db_url(url: &str) -> (String, Option<String>) {
+    let mut db_url = url::Url::parse(url).expect("Could not parse database url");
+    let mut cert_path = None;
+
+    let mut query = "".to_string();
+    db_url.query_pairs().for_each(|(k, v)| {
+        if k == "sslrootcert" {
+            cert_path = Some(v.parse().unwrap());
+        } else {
+            query.push_str(&format!("{}={}&", k, v));
+        }
+    });
+    db_url.set_query(Some(&query));
+
+    (db_url.to_string(), cert_path)
+}
+
+pub async fn new_db_pool(
+    database_url: &str,
+    max_pool_size: Option<u32>,
+) -> Result<ArcDbPool, PoolError> {
+    let (_url, cert_path) = parse_and_clean_db_url(database_url);
+
+    let config = if cert_path.is_some() {
+        let mut config = ManagerConfig::<MyDbConnection>::default();
+        config.custom_setup = Box::new(|conn| Box::pin(establish_connection(conn)));
+        AsyncDieselConnectionManager::<MyDbConnection>::new_with_config(database_url, config)
+    } else {
+        AsyncDieselConnectionManager::<MyDbConnection>::new(database_url)
+    };
+    let pool = Pool::builder()
+        .max_size(max_pool_size.unwrap_or(DEFAULT_MAX_POOL_SIZE))
+        .build(config)
+        .await?;
+    Ok(Arc::new(pool))
+}
+
+pub async fn execute_in_chunks<U, T>(
+    conn: ArcDbPool,
+    build_query: fn(Vec<T>) -> (U, Option<&'static str>),
+    items_to_insert: &[T],
+    chunk_size: usize,
+) -> Result<(), diesel::result::Error>
+where
+    U: QueryFragment<Backend> + diesel::query_builder::QueryId + Send + 'static,
+    T: serde::Serialize + for<'de> serde::Deserialize<'de> + Clone + Send + 'static,
+{
+    let tasks = items_to_insert
+        .chunks(chunk_size)
+        .map(|chunk| {
+            let conn = conn.clone();
+            let items = chunk.to_vec();
+            tokio::spawn(async move {
+                let (query, additional_where_clause) = build_query(items.clone());
+                execute_or_retry_cleaned(conn, build_query, items, query, additional_where_clause)
+                    .await
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let results = futures_util::future::try_join_all(tasks)
+        .await
+        .expect("Task panicked executing in chunks");
+    for res in results {
+        res?
+    }
+
+    Ok(())
+}
+
+pub async fn execute_with_better_error<U>(
+    pool: ArcDbPool,
+    query: U,
+    mut additional_where_clause: Option<&'static str>,
```

Comment:
let's get rid of this, just like what @CapCap is doing in https://github.com/aptos-labs/aptos-indexer-processors/pull/288?

Reply:
I'll track this in a task for now